### PR TITLE
K8s tests: revamp

### PR DESCRIPTION
**Pull request by robertomonteromiguel**
## Motivation

K8s Lib Injection tests

- Now the scenario will run the K8s Cluster.
- Delete unused code:
1.  Remove retry wrapper. Now we only run one k8s cluster per machine, we won't have conflicts with the kubectl configuration and open ports.
2. The test are not parametrized. We can remove the conftest and the parameter object.
3. Clean unused methods
- Introduce the "K8s provider" concept. We allow three provideers:
1.  **Kind Cluster**: Default provider. This provider is local managed, that it's mean its lifecycle is managed by the system-tests scenario (start and stop cluster from the system-tests scenario). 
2. **MiniKube**: Using a command parameter, you can run the tests using a MiniKube cluster. This provider is local managed. The system-tests will start and stop the cluster.
3. **AWS EKS**: Beta provider. It's not locally managed. You need to run the remote cluster manually
 and be sure that the Kubectl has the correct access.
- Update the documentation.
- Remove the tests from GitHub actions. We only keep the image validation scenarios and the weblog build process.
- Add the new K8s Lib Injection tests to Gitlab as part of the SSI pipeline

## Workflow

1. ⚠️ Create your PR as draft ⚠️
4. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
5. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Review by pawelchcki — APPROVED** [review]

**Comment by pawelchcki** [comment]
Nice! - The initial clluster setup seems a bit cleaner and more flexible.

The following configuration of the cluster itself - I still think is a little bit hard to follow, but I can't offer a better alternative for now :| 

Otherwise looks good to me !

**Review by cbeauchesne — APPROVED** [review]

**Comment by cbeauchesne** [comment]
Good job !
